inst/include/cpp11/protect.hpp

```diff
@@ -198,13 +198,15 @@ inline void check_user_interrupt() {
 }
 
 template <typename... Args>
-void stop(const char* fmt, Args... args) {
+void stop [[noreturn]](const char* fmt, Args... args) {
   unwind_protect([&] { Rf_error(fmt, args...); });
+  throw std::runtime_error("stop()");
```

Comment:
Will this ever be executed (for example if we don't have `R_UnwindProtect`)? If not, I'd replace this with an explanatory comment and a call to `std::abort()` (which should satisfy the compiler since it also has the required `[[noreturn]]` attribute)

Reply:
No, I don't believe it should never get executed, if we don't have `R_UnwindProtect` then `Rf_error()` will do a C long jump out of the function.

Another alternative would be `std::terminate()`, I don't have a strong feeling one way or the other, the main reason I suggested throwing an error rather than terminating is generally R packages should never cause the R process itself to terminate, and the runtime error should get converted to an R level error by the wrapping code.

Reply:
If it never gets called then the `abort()`/`throw`/... is only there to help the compiler; ideally `Rf_error` would be marked with `[[noreturn]]` which would allow the compiler to infer that no code path through `cpp11::stop` will return. IMHO this should be included as a comment
```suggestion
  // Compiler hint to allow [[noreturn]] attribute; this is never executed since Rf_error will longjmp
  throw std::runtime_error("stop()");
```
since readers unfamiliar with the attribute might expect the code to be run and wrap things in unnecessary `try/catch` blocks.